src/frequenz/client/reporting/_types.py

```diff
@@ -92,8 +92,8 @@ def __iter__(self) -> Iterator[MetricSample]:
         for item in items:
             cid = getattr(item, self.id_attr)
             for sample in getattr(item, "metric_samples", []):
-                ts = sample.sampled_at.ToDatetime().replace(tzinfo=timezone.utc)
-                met = Metric.from_proto(sample.metric).name
+                ts = sample.sample_time.ToDatetime().replace(tzinfo=timezone.utc)
+                met = Metric(sample.metric).name
```

Comment:
You can also import from `frequenz.client.common.proto` and use:

```suggestion
                ts = datetime_from_proto(sample.sample_time)
                met = enum_from_proto(Metric, sample.metric).name
```

The datetime conversion is the same, it only avoids the mistake of forgetting the timezone. But when using `enum_from_proto()` we could support multiple api-common versions in the future, maybe. And it can handle when the proto added a new metric but the client doesn't know about it yet. In this case, your code will raise an exception.

Same for other conversions like these.

Reply:
Good idea, updated.

Reply:
I see you use `allow_invalid=False`, which also will raise.

I guess it is on purpose, and I see maybe changing it now might not be trivial, but long term I think it would be better to keep clients forward-compatible by allowing the server returning some values we don't know about yet instead of raising an exception.

This will allow users to still get some metric the client doesn't support yet, if for some reason they are stuck with an old client version.

Reply:
Yes, it's on purpose to have minimal changes. There are long-standing plans of a major rework of the types once the whole microgrid/common updates are final and we go towards v1.